### PR TITLE
Test Stack Methods

### PR DESCRIPTION
**Pull request by brianwchou**
## Description

test each of the following:
- [x] peek()
- [x] push()
- [x] pop()

### CODE REVIEW

**Comment by ejwessel** [comment]
@brianwchou lets also add
https://www.npmjs.com/package/ganache-time-traveler

You can do this in another PR

File: test/Stack.test.js [hunk 1]
```diff
@@ -5,43 +5,46 @@ contract('Stack', async () => {
     
     var stack;
 
-    beforeEach(async () => {
```
Comment (ejwessel) [inline test/Stack.test.js:8]: add https://www.npmjs.com/package/ganache-time-traveler

**Review by ejwessel — CHANGES_REQUESTED** [review]
Waiting on https://github.com/brianwchou/Soldity-Stack-Example/pull/2

File: test/Stack.test.js [hunk 2]
```diff
@@ -5,43 +5,46 @@ contract('Stack', async () => {
...
-        assert.equal(value.toNumber(), 1, "value was not pushed")
+        assert.equal(value.toNumber(), 1, 'value was not pushed')
     });
 
-    it("getSize() returns length with stack empty", async () => {
-        let value = await stack.getSize()
-        assert.equal(value.toNumber(), 0, 'stack is not empty')
+    it('push() fails when stack is at capacity', async() => {
+        for (let i = 0; i < 10; i++) { await stack.push(5) }
+        truffleAssert.fails(stack.push(5))
```
Comment (ejwessel) [inline test/Stack.test.js]: use `reverts()`

You also need an await at the beginning of this line

File: test/Stack.test.js [hunk 3]
```diff
@@ -5,43 +5,46 @@ contract('Stack', async () => {
...
-
-    // it("getSize() returns length with stack filled", () => {
-
-    // })
-
-    // it("do something", () => {
-
-    // })
+    it('peek() reverts with stack empty', async() => {
+        truffleAssert.reverts(stack.peek(), 'stack needs to have a value')
```
Comment (ejwessel) [inline test/Stack.test.js]: you are missing an await at the beginning of this line.

**Review by ejwessel — CHANGES_REQUESTED** [review]

File: test/Stack.test.js (hunk 1 above)
Comment (brianwchou) [reply test/Stack.test.js:8]: Not sure how much this matters but tests run slightly faster without time traveler.
About 1-2ms for tests that have yellow or red times showing on test runner.

**Comment by ejwessel** [comment]
waiting for https://github.com/brianwchou/Soldity-Stack-Example/pull/8

File: test/Stack.test.js (hunk 1 above)
Comment (ejwessel) [reply test/Stack.test.js:8]: Good eye. At this point I'm ok with it.

File: test/Stack.test.js [hunk 4]
```diff
@@ -1,47 +1,56 @@
...
+    beforeEach(async() => {
+        let snapshot = await timeUtil.takeSnapshot()
+        snapshotId = snapshot.result
+    })
+
+    afterEach(async() => {
+        await timeUtil.revertToSnapShot(snapshotId)
+    })
+
+    it('push() increments size of stack', async() => {
```
Comment (ejwessel) [inline test/Stack.test.js:23]: Can we have two describe blocks?
- the first one will test initial state
- the second one will test setting state

**Review by ejwessel — CHANGES_REQUESTED** [review]
requesting.

File: test/Stack.test.js (hunk 4 above)
Comment (ejwessel) [reply test/Stack.test.js:23]: ignore this. let's do this in another pr

**Review by ejwessel — APPROVED** [review]
LGTM

**Pull request merged**